src/runtime-rs/crates/shim/src/shim_run.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::os::unix::io::RawFd;
+
+use anyhow::{Context, Result};
+
+use crate::{
+    logger,
+    shim::{ShimExecutor, ENV_KATA_BIND_FD},
+    Error,
+};
+
+impl ShimExecutor {
+    pub async fn run(&mut self) -> Result<()> {
+        crate::panic_hook::set_panic_hook();
+        let sid = self.args.id.clone();
+        let bundle_path = self.get_bundle_path().context("get bundle")?;
+        let path = bundle_path.join("log");
+        let _logger_guard =
+            logger::set_logger(path.to_str().unwrap(), &sid, self.args.debug).context("set logger");
+
+        self.do_run()
+            .await
+            .map_err(|err| {
+                error!(sl!(), "failed run shim {:?}", err);
+                err
+            })
+            .context("long run")?;
```

Comment:
`long run` is hard to understand.

Reply:
Thanks for the review, would it be more appropriate to use `run shim` as context